front/src/reducers/osrdconf/stdcmConf/index.ts

```diff
@@ -150,11 +151,34 @@ export const stdcmConfSlice = createSlice({
       state.timetableID = action.payload.timetableID;
       state.electricalProfileSetId = action.payload.electricalProfileSetId;
       state.searchDatetimeWindow = action.payload.searchDatetimeWindow;
-      if (action.payload.workScheduleGroupId) {
-        state.workScheduleGroupId = action.payload.workScheduleGroupId;
-      }
-      if (action.payload.temporarySpeedLimitGroupId) {
-        state.temporarySpeedLimitGroupId = action.payload.temporarySpeedLimitGroupId;
+      state.workScheduleGroupId = action.payload.workScheduleGroupId;
+      state.temporarySpeedLimitGroupId = action.payload.temporarySpeedLimitGroupId;
+
+      // check that the arrival dates are in the search time window
+      const { searchDatetimeWindow } = action.payload;
+      if (searchDatetimeWindow) {
+        const origin = state.stdcmPathSteps.at(0)! as Extract<StdcmPathStep, { isVia: false }>;
+        const destination = state.stdcmPathSteps.at(-1)! as Extract<
+          StdcmPathStep,
+          { isVia: false }
+        >;
```

Comment:
If you cast, I don't think you need the `!`

```suggestion
        const origin = state.stdcmPathSteps.at(0) as Extract<StdcmPathStep, { isVia: false }>;
        const destination = state.stdcmPathSteps.at(-1) as Extract<
          StdcmPathStep,
          { isVia: false }
        >;
```

Reply:
Done